### PR TITLE
Add garbage collection for inactive meters.

### PR DESCRIPTION
**Pull request by jjzeng-seattle**

### CODE REVIEW

File: metrics/resource_view_test.go [hunk 1]
```diff
@@ -166,6 +166,90 @@ func TestResourceAsString(t *testing.T) {
...
+		},
+	}
+
+	for _, tc := range testCases {
+		removeInactiveMeters(tc.inactiveThreshold, tc.metersForTest)
+		if len(tc.metersForTest.meters) != tc.remainingMeters {
+			t.Fatalf("Expect %d remaining meters, but got %d", tc.remainingMeters, len(tc.metersForTest.meters))
+		}
+	}
+	timeNow = time.Now
```
Comment (yanweiguo) [inline metrics/resource_view_test.go:220]: This line won't be reached if `t.Fatalf` happens before. I think we want to use a defer.

Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Thanks.  Still trying to get better with golang :-)

File: metrics/resource_view.go [hunk 2]
```diff
@@ -61,6 +62,40 @@ var allMeters = meters{
...
+func init() {
+	go garbageCollectMeters(10*time.Minute, time.Hour)
+}
+
+func garbageCollectMeters(frequency time.Duration, inactiveThreshold time.Duration) {
+	for {
+		time.Sleep(frequency)
+		removeInactiveMeters(inactiveThreshold, &allMeters)
+	}
+}
```
Comment (vagababov) [inline metrics/resource_view.go:77]: I'd rather not have this kind of code in the init by default.

File: metrics/resource_view.go [hunk 3]
```diff
@@ -61,6 +62,40 @@ var allMeters = meters{
...
+// A way to mock time.Now in test
+var timeNow = time.Now
+
+func init() {
+	go garbageCollectMeters(10*time.Minute, time.Hour)
+}
+
+func garbageCollectMeters(frequency time.Duration, inactiveThreshold time.Duration) {
+	for {
+		time.Sleep(frequency)
```
Comment (vagababov) [inline metrics/resource_view.go:74]: you want 
```
for _ = range time.Tick(duration) {
...
}
```
k8s clock would be helpful here as well.

File: metrics/resource_view.go [hunk 4]
```diff
@@ -61,6 +62,40 @@ var allMeters = meters{
 	meters: map[string]*meterExporter{"": &defaultMeter},
 }
 
+// A way to mock time.Now in test
+var timeNow = time.Now
```
Comment (vagababov) [inline metrics/resource_view.go:66]: I'd use k8s Clock type instead.

File: metrics/resource_view_test.go [hunk 5]
```diff
@@ -165,6 +166,90 @@ func TestResourceAsString(t *testing.T) {
...
+	testNow := time.Date(2050, time.August, 7, 9, 0, 0, 0, time.UTC)
+	timeNow = func() time.Time { return testNow }
+	testCases := []struct {
+		name              string
+		metersForTest     *meters
+		inactiveThreshold time.Duration
+		remainingMeters   int
+	}{
+		{
+			name: "all inactive",
```
Comment (vagababov) [inline metrics/resource_view_test.go:180]:
```suggestion
	}{{
			name: "all inactive",
```

File: metrics/resource_view_test.go [hunk 6]
```diff
@@ -165,6 +166,90 @@ func TestResourceAsString(t *testing.T) {
...
+				meters: map[string]*meterExporter{
+					"": {m: &defaultMeterImpl{}, t: timeNow()},
+					// using defaulteMeterImple as we don't want to add real meters to global states.
+					"foo": {m: &defaultMeterImpl{}, t: timeNow().Add(-2 * time.Hour)},
+					"bar": {m: &defaultMeterImpl{}, t: timeNow().Add(-time.Hour - time.Microsecond)},
+				}},
+			inactiveThreshold: time.Hour,
+			remainingMeters:   1, // we don't delete the default one
+		},
+		{
```
Comment (vagababov) [inline metrics/resource_view_test.go:191]:
```suggestion
			remainingMeters:   1, // we don't delete the default one
		},{
```
etc elsewhere

File: metrics/resource_view.go [hunk 7]
```diff
@@ -61,6 +62,40 @@ var allMeters = meters{
...
+
+func init() {
+	go garbageCollectMeters(10*time.Minute, time.Hour)
```
Comment (vagababov) [inline metrics/resource_view.go:69]: you want the time to be configurable/var for simpler testing as well.

**Pull request closed without merging**